### ISSUE TITLE
Excluding primary key from api view breaks POST

### ISSUE THREAD

**Issue by graup** [jfinkels/flask-restless#400]
I tried excluding the primary key in `create_api` as mentioned here:

http://flask-restless.readthedocs.org/en/latest/customizing.html#specifying-which-columns-are-provided-in-responses

However, this breaks POSTing to this view, as the post handler tries to get the pk from the database object after inserting.

```
File "flask_restless/views.py", line 1390, in post
    primary_key = result[pk_name]
KeyError: 'id'
```

I was going to submit a failing unit test, but upon further thinking it actually makes sense that this doesn't work. After all, how should a client know which instance it created if the response may not contain the id?

So I guess, it would just be useful to add a hint to the docs explaining this. Something like "Make sure the primary key field is included if you intend to allow POST".

**Comment by jfinkels** [comment]
This should definitely be in the documentation, but it's also a bug in the code. The code should raise an error and inform the user that he or she is not allowed to exclude primary keys.

**Comment by graup** [comment]
I would add: that he or she is not allowed to exclude primary keys, as long as other methods than GET are allowed. It might make sense to some users to not expose keys on a read only API.

If you don't mind I'd try coming up with a pull request for raising the error (incl. unit test).

**Comment by jfinkels** [comment]
I have merged in a fix for this at 3fd6c8af9f2d802a59e02658c54484769ab21dcd.
